pkg/chunk/cache/fifo_cache.go

```diff
@@ -33,7 +33,7 @@ var (
 		Help:      "The total number of evicted entries",
 	}, []string{"cache"})
 
-	cacheEntriesCurrent = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	cacheEntriesCurrent = promauto.NewCounterVec(prometheus.CounterOpts{
```

Comment:
The fact that currently the number of items in the cache can't decrease is just an implementation detail. Being an LRU cache and having the concept of eviction, in the future we may actually have a routine deleting expired items. For this reason, I would suggest to revert this back to a gague.

Reply:
agree!